modules/auxiliary/scanner/http/cisco_device_manager.rb

```diff
@@ -54,9 +59,35 @@ def run_host(ip)
 
     if res and res.body and res.body =~ /Cisco (Internetwork Operating System|IOS) Software/
       print_good("#{rhost}:#{rport} Successfully authenticated to this device")
+      service_data = {
+        address: rhost,
+        port: rport,
+        service_name: 'http',
+        protocol: 'tcp',
+        workspace_id: myworkspace_id
+      }
+
+      credential_data = {
+        module_fullname: self.fullname,
+        origin_type: :service,
+        private_data: datastore['HttpPassword'],
+        private_type: :password,
+        username: datastore['HttpUsername']
+      }
+
+      credential_data.merge!(service_data)
+      credential_core = create_credential(credential_data)
+      login_data = {
+        core: credential_core,
+        last_attempted_at: DateTime.now,
+        status: Metasploit::Model::Login::Status::SUCCESSFUL
+      }
+      login_data.merge!(service_data)
+      create_credential_login(login_data)
```

Comment:
Since this includes `HttpClient` this whole addition can be replaced with:
```
store_valid_credential(datastore['HttpUsername'], datastore['HttpPassword'])
```

Reply:
that is much, MUCH easier.